web-ui/src/helpers/query-parameters.js

```diff
@@ -0,0 +1,56 @@
+import { useEffect } from 'react';
+
+/**
+ * @typedef {object} QPBoolean
+ * @property {string} name
+ * @property {boolean} default
+ * @property {boolean} value
+ * @property {(boolean) => void} setter - takes query parameter value and updates state
+ * @property {(any) => string} toQP - takes state value and returns query parameter value
+ */
+
+/**
+ * @typedef {object} QPString
+ * @property {string} name
+ * @property {string} default
+ * @property {string} value
+ * @property {(string) => void} setter - takes query parameter value and updates state
+ * @property {(any) => string} toQP - takes state value and returns query parameter value
+ */
+
+/**
+ * @param {(QPBoolean | QPString[]} qps - query parameters
+ */
+export const queryParameterSetup = qps => {
```

Comment:
Should we consider this a [custom hook](https://react.dev/learn/reusing-logic-with-custom-hooks), and start the name with `useQueryParameterSetup` to make it clear this function is intended only for use within react components by way of convention? I think this might qualify...

Reply:
I did a little more digging, and custom hooks are intended for encapsulating React state. Here we're not encapsulating state, and so then I think this would not qualify as a custom hook. Though I'm not 100%.

Reply:
Interesting idea. I suppose not all hooks encapsulate state. For example, this does not:

```js
useEffect(() => {
  console.log('page was loaded');
}, []);
```

Reply:
I did decide to rename `queryParameterSetup` to `useQueryParameters` since it does call `useEffect`. Great suggestions!